content/learn/book/welcome/setup/_index.md

```diff
@@ -95,6 +97,81 @@ Now run `cargo run` again. The Bevy dependencies should start building. This wil
 
 Now that we have our Bevy project set up, we're ready to start making our first Bevy app!
 
+### Checking that it works
+
+Within `main.rs`, let's create our first app and check that all the dependencies are working correctly!
+
+```rust
+use bevy::prelude::*;
+
+fn main(){
+  App::build().add_plugins(DefaultPlugins).run();
+}
+```
+
+Use `cargo run` to test it out.
+This will just show an empty window, and includes all of the standard functionality needed to make Bevy games.
+
+### Performance by default
+
+Rust's compiler supports several levels of performance optimization, with fairly dramatic effects on Bevy's performance.
+For a more pleasant and realistic development experience, we recommend enabling optimized compilation by default by adding the following lines to your `Cargo.toml`.
+
+```toml
+[profile.dev]
+opt-level = 3
+```
+
+### Faster compiling
+
+* **Bevy's Dynamic Linking Feature**: This is the most impactful iterative compilation time decrease! It requires no special setup except on Windows, where you also have to follow the three numbered steps below for it to work. If `bevy` is a dependency you can compile the binary with the `dynamic` feature flag (enables dynamic linking):
+
+  ```sh
+  cargo run --features bevy/dynamic
+  ```
+
+  If you don't want to add the `--features bevy/dynamic` to each run, this flag can permanently be set via `Cargo.toml`:
+
+  ```toml
+  [dependencies]
+  bevy = { version = "0.5.0", features = ["dynamic"] }
+  ```
+
+  NOTE: Remember to revert this before releasing your game! Otherwise you will need to include `libbevy_dylib` alongside your game if you want it to run. If you remove the `dynamic` feature, your game executable can run standalone.
+
+For the fastest iterative compile times, we recommend the following configuration as well. You need to follow all three steps for any effect:
+
+1. **LLD linker**: The Rust compiler spends a lot of time in the "link" step. LLD is _much faster_ at linking than the default Rust linker. To install LLD, find your OS below and run the given command:
+   * **Ubuntu**: `sudo apt-get install lld clang`
+   * **Arch**: `sudo pacman -S lld`
+   * **Windows**: Ensure you have the latest [cargo-binutils](https://github.com/rust-embedded/cargo-binutils)
+
+      ```sh
+      cargo install -f cargo-binutils
+      rustup component add llvm-tools-preview
+      ```
+
+   * **MacOS**: Modern LLD does not yet support MacOS, but we can use zld instead: `brew install michaeleisel/zld/zld`
+2. **Nightly Rust Compiler**: This gives access to the latest performance improvements and "unstable" optimizations
```

Comment:
I think this step should clearly be marked as optional or at least be more obvious that this is purely for compilation speed and that bevy doesn't require nightly to work. Bevy works without any issues on stable rust so I wouldn't want anyone to think they _need_ nightly.